lib/ansible/modules/cloud/amazon/cloudfront_origin_access_identity.py

```diff
@@ -0,0 +1,285 @@
+                    'Comment': comment
+                },
+                Id=origin_access_identity_id, IfMatch=e_tag)
+        except botocore.exceptions.ClientError as e:
+            self.module.fail_json(
+                msg="error updating cloud front origin access identity - " +
+                str(e) + "\n" + traceback.format_exc())
+
+
+class CloudFrontOriginAccessIdentityValidationManager(object):
+    """
+    Manages Cloudfront Origin Access Identities
+    """
+
+    def __init__(self, module):
+        self.module = module
+        self.__cloudfront_facts_mgr = CloudFrontFactsServiceManager(module)
+
+    def validate_etag_from_origin_access_identity_id(self,
+                                                     origin_access_identity_id):
+        try:
+            if origin_access_identity_id is None:
+                return
+            oai = self.__cloudfront_facts_mgr.get_origin_access_identity(
+                origin_access_identity_id)
+            if oai is not None:
+                return oai.get('ETag')
+        except Exception as e:
+            self.module.fail_json(
+                msg="error getting etag from origin_access_identity_id - " +
+                str(e) + "\n" + traceback.format_exc())
+
+    def validate_origin_access_identity_id_from_caller_reference(
+            self, caller_reference):
+        try:
+            origin_access_identities = (
+                self.__cloudfront_facts_mgr.list_origin_access_identities())
+            origin_origin_access_identity_ids = [oai.get('Id') for oai in
+                                                 origin_access_identities]
+            for origin_access_identity_id in origin_origin_access_identity_ids:
+                oai_config = (
+                    self.__cloudfront_facts_mgr.get_origin_access_identity_config(
+                        origin_access_identity_id))
+                temp_caller_reference = oai_config.get(
+                    'CloudFrontOriginAccessIdentityConfig').get(
+                    'CallerReference')
+                if temp_caller_reference == caller_reference:
+                    return origin_access_identity_id
+        except Exception as e:
+            self.module.fail_json(
+                msg="error getting origin access identity from " +
+                "caller_reference - " + str(e) + "\n" + traceback.format_exc())
+
+    def validate_comment(self, comment):
+        if comment is None:
+            comment = (
+                "origin access identity created by Ansible with datetime " +
+                datetime.datetime.now().strftime('%Y-%m-%dT%H:%M:%S.%f'))
+        return comment
+
+
+def main():
+    argument_spec = ec2_argument_spec()
+
+    argument_spec.update(dict(
+        state=dict(choices=['present', 'absent'], default='present'),
+        origin_access_identity_id=dict(required=False, default=None,
+                                       type='str'),
```

Comment:
`type='str'` is the default so you can remove that. `default=None` and `required=False` are also defaults that can be removed.